tests/lib/__init__.py

```diff
@@ -46,7 +46,9 @@
     # Literal was introduced in Python 3.8.
     from typing import Literal
 
-    ResolverVariant = Literal["resolvelib", "legacy"]
+    ResolverVariant = Literal[
+        "resolvelib", "legacy", "2020-resolver", "legacy-resolver"
+    ]
```

Comment:
I think it'd be better to modify the code paths to enforce the "resolvelib" and "legacy" values consistently instead.

Reply:
I made this change locally, but I'm a little less sure of it. Maybe better to do in a separate PR?

Reply:
Cool! As long as you don't forget to file it. 🙃

Reply:
It's a deal